pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -963,7 +973,106 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1.Mac
 		return &clustererror.RequeueAfterError{RequeueAfter: time.Second * 5}
 	}
 
-	// node is now available again
+	// node is now available again - restore annotations and labels
+	if _, annotationsBackupExists := machine.Annotations[nodeAnnotationsBackupAnnotation]; annotationsBackupExists {
+		return a.restoreAnnotationsAndLabels(node, machine, ctx)
+	}
+
+	//remediation is done
 	log.Printf("Node %s is available, remediation of Machine %s complete", node.Name, machine.Name)
 	return a.deleteRemediationAnnotations(ctx, machine)
 }
+
+// storeAnnotationsAndLabels copies node's annotations and labels and stores them on machine's annotations
+func (a *Actuator) storeAnnotationsAndLabels(node *corev1.Node, machine *machinev1.Machine, ctx context.Context) error {
+	marshaledAnnotations, err := marshal(node.Annotations)
+	if err != nil {
+		log.Printf("Failes to marshal node %s annotations assoicated with Machine %s: %s",
+			node.Name, machine.Name, err.Error())
+		return err
+	}
+
+	marshaledLabels, err := marshal(node.Labels)
+	if err != nil {
+		log.Printf("Failes to marshal node %s labels assoicated with Machine %s: %s",
+			node.Name, machine.Name, err.Error())
+		return err
+	}
+
+	machine.Annotations[nodeLabelsBackupAnnotation] = marshaledLabels
+	machine.Annotations[nodeAnnotationsBackupAnnotation] = marshaledAnnotations
+
+	err = a.client.Update(ctx, machine)
+	if err != nil {
+		log.Printf("failed to update machine with node's annotations and labels %s: %s", machine.Name, err.Error())
+		return err
+	}
+
+	return &clustererror.RequeueAfterError{}
+}
+
+// restoreAnnotationsAndLabels copies annotations and labels stored in machine's annotation to the node
+// and removes the annotations used to store that data
+func (a *Actuator) restoreAnnotationsAndLabels(node *corev1.Node, machine *machinev1.Machine, ctx context.Context) error {
+	if len(machine.Annotations) == 0 {
+		return &clustererror.RequeueAfterError{}
+	}
+
+	nodeAnn, err := unmarshal(machine.Annotations[nodeAnnotationsBackupAnnotation])
+	if err != nil {
+		log.Printf("failed to unmarshal node's annotations from %s: %s", machine.Name, err.Error())
+		return err
+	}
+
+	nodeLabels, err := unmarshal(machine.Annotations[nodeLabelsBackupAnnotation])
+	if err != nil {
+		log.Printf("failed to unmarshal node's labels from %s: %s", machine.Name, err.Error())
+		return err
```

Comment:
Likewise, I don't think we should go into a never ending loop if our stored labels are corrupted.
Maybe we should remove bad values?

Reply:
Not sure either. Maybe this should be configurable? 
For one admin, annotations/labels might be critical, such that having a new machine without preserving annotations/labels makes more harm than good?
And for the other - it might be just best effort.

Reply:
I'll suggest we just don't re-try in that case. There's no reason to think this will succeed until someone either fixes the data or fixes the software. Either one will cause reconciliation to happen again.